src/components/PupilComponents/QuizOrderAnswer/QuizOrderAnswer.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  OakBox,
+  OakDraggableFeedback,
+  OakDroppable,
+  OakQuizOrder,
+  OakQuizOrderProps,
+} from "@oaknational/oak-components";
+import { useMemo, useState } from "react";
+import { isArray } from "lodash";
+
+import { isOrderAnswer } from "../QuizUtils/answerTypeDiscriminators";
+
+import { useQuizEngineContext } from "@/components/PupilComponents/QuizEngineProvider";
+import { useInitialChange } from "@/components/PupilComponents/QuizUtils/useInitialChange";
+import OakError from "@/errors/OakError";
+
+export type QuizOrderAnswerProps = {
+  onInitialChange?: () => void;
+  onChange?: () => void;
+};
+
+export const QuizOrderAnswer = ({
+  onChange,
+  onInitialChange,
+}: QuizOrderAnswerProps) => {
+  const { handleOnChange: handleInitialChange } = useInitialChange({
+    onChange,
+    onInitialChange,
+  });
+  const { currentQuestionData, questionState, currentQuestionIndex } =
+    useQuizEngineContext();
+  invariant(currentQuestionData, "currentQuestionData is not defined");
+  const answers = currentQuestionData.answers;
+  const questionUid = currentQuestionData.questionUid;
+  const feedback = questionState[currentQuestionIndex]?.feedback;
+
+  invariant(
```

Comment:
I'd prefer this guard were pushed to the edges but the GQL types are pretty loose with nulls. The page won't be usable without these so we should just blow up that way we can at least identify that the lesson is broken in logs rather than trying to render something meaningless through optional chaining.

Reply:
Unfortunately the schema is shared with teacher sq and if we tightened it we'd probably break their code but agreed nullable is way overused.